rotkehlchen/tests/exchanges/test_bitfinex.py

```diff
@@ -1499,3 +1637,76 @@ def mock_api_query_response(endpoint, options):  # pylint: disable=unused-argume
             ),
         ]
         assert asset_movements == expected_asset_movements
+
+
+def test_partial_query_online_history_events(mock_bitfinex: 'Bitfinex') -> None:
+    """Test that the logic for history events works as expected when there is an error
+    causing only part of the range to be queried.
+    """
+    api_limit = 2
+    mock_bitfinex.first_connection = MagicMock()  # type: ignore
+    mock_bitfinex.pair_bfx_symbols_map = {'ETHUST': ('ETH', 'UST')}
+    expected_calls = [
+        call(
+            endpoint='movements',
+            options={
+                'start': 0,
+                'end': 1743533561000,
+                'limit': 2,
+            },
+        ),
+        call(
+            endpoint='trades',
+            options={
+                'start': 0,
+                'end': 1743533561000,
+                'limit': 2,
+                'sort': 1,
+            },
+        ),
+        call(
+            endpoint='trades',
+            options={
+                'start': 1606901400000,
+                'end': 1743533561000,
+                'limit': 2,
+                'sort': 1,
+            },
+        ),
+    ]
+
+    def get_paginated_response() -> Generator[tuple[HTTPStatus, str], None, None]:
+        results = [
+            (HTTPStatus.OK, f'[{MOVEMENT_1}]'),
+            (HTTPStatus.OK, f'[{TRADE_1},{TRADE_2}]'),
+            (HTTPStatus.INTERNAL_SERVER_ERROR, 'xxxxxxx'),
+        ]
+        yield from results
+
+    def mock_api_query_response(endpoint: str, options: dict[str, Any]) -> MockResponse:  # pylint: disable=unused-argument
+        status, text = next(get_response)
+        return MockResponse(status, text)
+
+    get_response = get_paginated_response()
+    with ExitStack() as stack:
+        stack.enter_context(patch(
+            target='rotkehlchen.exchanges.bitfinex.API_MOVEMENTS_MAX_LIMIT',
+            new=api_limit,
+        ))
+        stack.enter_context(patch(
+            target='rotkehlchen.exchanges.bitfinex.API_TRADES_MAX_LIMIT',
+            new=api_limit,
+        ))
+        api_query_mock = stack.enter_context(patch.object(
+            target=mock_bitfinex,
+            attribute='_api_query',
+            side_effect=mock_api_query_response,
+        ))
+        events, actual_end_ts = mock_bitfinex.query_online_history_events(
+            start_ts=Timestamp(0),
+            end_ts=(end_ts := Timestamp(1743533561)),
+        )
+        assert api_query_mock.call_args_list == expected_calls
+        assert len(events) == 8  # withdrawal/fee from one movement, and spend/receive/fee from two trades.  # noqa: E501
+        assert actual_end_ts == events[-1].timestamp / 1000
```

Comment:
nitpick use ts_sec_to_ms here

```suggestion
        assert actual_end_ts == events[-1].timestamp / 1000
```